pkg/catalogv2/system/system.go

```diff
@@ -84,14 +93,27 @@ func NewManager(ctx context.Context,
 func (m *Manager) Start(ctx context.Context) {
 	m.ctx = ctx
 	go m.runSync()
+
+	m.settings.OnChange(ctx, "system-feature-chart-refresh", m.onSetting)
```

Comment:
I forget, these all fire on startup yes ? Looks like it starts the interval change below, and that should fire it off the bat and then wait the interval ?

Reply:
This doesn't fireup on restart though. On restart, everything will be empty(since it is memory) and controller kicks in to add desiredCharts and then manager will reinstall/upgrade all the charts. This is just for watching the settings so that when setting changes, we change the time ticker here. https://github.com/rancher/rancher/pull/33879/files/90c2409439f32318c0552a5836a83c696fa75f36#diff-96f7551d1bcafc092b3502857d33f73f576c753e3a4311b8660fabc446b383b0R115-R116

Reply:
This was wrong line to add that comment. So the manager `runSync` method down there on line 109 is what will fire at startup.